### ISSUE TITLE
Unhandled Elf Relocation: Type = 42 (0x2a) Symbol = .text (0x2). ARM:BE:32:v8 (1.102)

### ISSUE THREAD

**Issue by pabx06** [NationalSecurityAgency/ghidra#2276]
**Describe the bug**
A clear and concise description of the bug.
xmlrole.o ( ARM:BE:32:v8 (1.102)) from libexpat ar archive 
![image](https://user-images.githubusercontent.com/4013370/93114777-54307300-f6bb-11ea-82d5-d0dbe71c1a30.png)
XREF by:
![image](https://user-images.githubusercontent.com/4013370/93114835-66121600-f6bb-11ea-8bea-c83c2d7646ec.png)

**To Reproduce**
compiled few libs  with an armeb tool-chain.
batch loading libexpat.a from this build release
with CFLAGS="-fPIC -ggdb -Og -fno-eliminate-unused-debug-symbols -g3   -fno-eliminate-unused-debug-types"
```
wget https://github.com/pabx06/arm-lib/releases/download/v1.0.4/arm-lib.tgz
tar xzvf  arm-lib.tgz ./install/libdir/libexpat.a
```

**Expected behavior**
i think that some XREF that link strings  <=> instruction/functions might be missing.
the main intent of loading that lib is to locate data constants with their functions / signature kinda like FID based on data referenced instead of hashing bit-masked instructions 

**Attachments**
the libexpat.a files is attached here:
[libexpat.a.gz](https://github.com/NationalSecurityAgency/ghidra/files/5219904/libexpat.a.gz)

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04.4 LTS 
 - Java Version: oracle 14
 - Ghidra 7e5f015d651b9c5e7825cb10bc50338b5af67538 with some change in the look and feel and swing colors

**Comment by pabx06** [comment]
here is an example of missing xref:
original source code notice `static const char * const types[] ` static initial content 
![image](https://user-images.githubusercontent.com/4013370/93119383-ffdcc180-f6c1-11ea-8d1f-a02b6b63f60e.png)

notice the highlited address in the decompile view `&DAT_000111f6` and `attlist2::lexical_block_0::types` in the listing
![image](https://user-images.githubusercontent.com/4013370/93119614-6bbf2a00-f6c2-11ea-85d9-bac76d781393.png)

i woul expecte that  `attlist2::lexical_block_0::types` would have in xref to `int attlist2(PROLOG_STATE *state,         int tok,         const char *ptr,         const char *end,      const ENCODING *enc)`

**Comment by ghidra1** [comment]
I will look into support for adding this relocation type.

**Comment by ghidra1** [comment]
There also appears to be an issue with processing of relocation type 3 (R_ARM_REL32) which is actually more of a problem.  I will be fixing this first then look into the type 42 (R_ARM_PREL31).

**Comment by astrelsky** [comment]
> There also appears to be an issue with processing of relocation type 3 (R_ARM_REL32) which is actually more of a problem. I will be fixing this first then look into the type 42 (R_ARM_PREL31).

There are a few issues with arm relocations. #2263 fixes R_ARM_ABS32

**Comment by ghidra1** [comment]
While I am implementing the R_ARM_PREL31 relocation I don't believe it is of high value within these object modules since they are all contained within the `.ARM.exidx` section which relates to stack unwinding.

**Comment by ghidra1** [comment]
Fixed in master branch for 9.2 release

**Comment by pabx06** [comment]
problem solved
![image](https://user-images.githubusercontent.com/4013370/93539985-98897080-f952-11ea-8216-c8bf4da51348.png)
